bundles/binding/org.openhab.binding.denon/src/main/java/org/openhab/binding/denon/internal/DenonGenericBindingProvider.java

```diff
@@ -0,0 +1,117 @@
+/**
+ * Copyright (c) 2010-2015, openHAB.org and others.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.denon.internal;
+
+import java.util.Map.Entry;
+import java.util.regex.Pattern;
+
+import org.openhab.binding.denon.DenonBindingProvider;
+import org.openhab.core.binding.BindingConfig;
+import org.openhab.core.items.Item;
+import org.openhab.core.items.ItemRegistry;
+import org.openhab.core.library.items.DimmerItem;
+import org.openhab.core.library.items.StringItem;
+import org.openhab.core.library.items.SwitchItem;
+import org.openhab.model.item.binding.AbstractGenericBindingProvider;
+import org.openhab.model.item.binding.BindingConfigParseException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This class is responsible for parsing the Denon binding configuration.
+ * 
+ * Some valid binding configuration strings:
+ * 
+ * {denon="avr2000#PW"} - Power on/off
+ * {denon="avr2000#SURROUNDMODE"} - Current surround mode
+ * 
+ * @author Jeroen Idserda
+ * @since 1.7.0
+ */
+public class DenonGenericBindingProvider extends AbstractGenericBindingProvider implements DenonBindingProvider {
+	
+	private static final Logger logger = LoggerFactory.getLogger(DenonGenericBindingProvider.class);	
+	
+	// Example: avr2000#PW, avr3000#SURROUNDMODE
+	private static final Pattern CONFIG_PATTERN = Pattern.compile("^(.)+#(.)+$");
+	
+	private ItemRegistry itemRegistry;
```

Comment:
please move the ItemRegistry into the Binding since it is not really used here but only referred to